### PR TITLE
Update aws-sdk-go and change way to get regional sts endpoint

### PR DESCRIPTION
**Pull request by jaydeokar**
*Issue #, if available:*
N/A

*Description of changes:*
This updates the aws-sdk-go and the way to get regional sts endpoint

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jayanthvn** [comment]
Need to fix -

```
go vet ./...
# github.com/aws/amazon-vpc-resource-controller-k8s/pkg/utils
# [github.com/aws/amazon-vpc-resource-controller-k8s/pkg/utils]
Error: vet: pkg/utils/helper_test.go:545:20: assignment mismatch: 3 variables but GetSourceAcctAndArn returns 4 values
make: *** [Makefile:30: verify] Error 1
```

File: pkg/aws/ec2/api/wrapper.go [hunk 1]
```diff
@@ -892,3 +892,35 @@ func (e *ec2Wrapper) DisassociateTrunkInterface(input *ec2.DisassociateTrunkInte
 	}
 	return err
 }
+
+func (e *ec2Wrapper) getRegionalStsEndpoint(partitionID, region string) (endpoints.ResolvedEndpoint, error) {
```
Comment (jayanthvn) [inline pkg/aws/ec2/api/wrapper.go:896]: We don't have UTs for this?

Comment (jaydeokar) [reply pkg/aws/ec2/api/wrapper.go:896]: Added

File: pkg/aws/ec2/api/wrapper.go [hunk 2]
```diff
@@ -892,3 +892,35 @@ func (e *ec2Wrapper) DisassociateTrunkInterface(input *ec2.DisassociateTrunkInte
...
+	var partition *endpoints.Partition
+	var stsServiceID = "sts"
+	for _, p := range endpoints.DefaultPartitions() {
+		if partitionID == p.ID() {
+			partition = &p
+			break
+		}
+	}
+	if partition == nil {
+		return endpoints.ResolvedEndpoint{}, fmt.Errorf("partition %s not valid", partitionID)
```
Comment (haouc) [inline pkg/aws/ec2/api/wrapper.go:906]: nit: ideally we should return `nil, err` when error is not nil though it is ok to leave it as is for now.

**Review by haouc — APPROVED** [review]
lgtm

File: pkg/aws/ec2/api/wrapper.go (hunk 2 above)
Comment (jaydeokar) [reply pkg/aws/ec2/api/wrapper.go:906]: Ack, I'll update this as a follow up

**Pull request merged**